merlin/util/snakewriter.py

```diff
@@ -66,18 +77,24 @@ def _generate_shell(self) -> str:
             shellString = 'python '
         else:
             shellString = self._clean_string(self._pythonPath) + ' '
+        shellString += '-m merlin '
+        shellString += self._clean_string(
+            self._analysisTask.dataSet.dataSetName) + ' '
+
+
         shellString += ''.join(
-            ['-m merlin -t ',
-             self._clean_string(self._analysisTask.analysisName),
-             ' -e \"',
-             self._clean_string(self._analysisTask.dataSet.dataHome), '\"',
-             ' -s \"',
-             self._clean_string(self._analysisTask.dataSet.analysisHome),
-             '\"'])
+            ['-t ', self._clean_string(self._analysisTask.analysisName),
+             ' -e \"', self._clean_string(self._analysisTask.dataSet.dataHome),
+             '\"', ' -s \"',
+             self._clean_string(self._analysisTask.dataSet.analysisHome), '\"'])
+        if isinstance(self._analysisTask.dataSet, dataset.MetaMERFISHDataSet):
+            shellString += ' --dataset-class \"MetaMERFISHDataSet\"'
+            shellString += ' --contained-datasets '
```

Comment:
After storing the list of datasets, --contained-datasets shouldn't be specified by snakemake for running every task.